yb-voyager/cmd/importData.go

```diff
@@ -1143,6 +1143,11 @@ func beforeIndexCreation(sqlInfo sqlInfo, conn **pgx.Conn, objType string) error
 		return fmt.Errorf("extract qualified index name from DDL [%v]: %w", sqlInfo.stmt, err)
 	}
 	if invalidTargetIndexesCache == nil {
+		tdb = tgtdb.NewTargetDB(&tconf)
```

Comment:
@sanyamsinghal because I moved indexes to the import-schema by default, I have to create targetDB here. I can't create it at the beginning of import-schema, because Init checks if the schema is present, and at the time, the schema objects haven't yet been created. 

I'm not too happy about creating targetDB here, because it seems out of place. Any suggestions? 
One approach I was thinking of was: let's not have `InvalidIndexes` be part of target_db_interface. Let us use the connection object that is part of this function to determine the invalid indexes. Right now, we don't really care about importing schema/indexes to source-replicas; it's only YB we care about. 
cc: @priyanshi-yb

Reply:
Agreed, for the approach having this `InvalidIndexes` not in target_db_interface. As for only `import-schema` we need this and it is not a hard requirement to be a part of target_db_iterface.